plugins/BEdita/Core/src/Shell/BeditaShell.php

```diff
@@ -0,0 +1,217 @@
+<?php
+/**
+ * BEdita, API-first content management framework
+ * Copyright 2016 ChannelWeb Srl, Chialab Srl
+ *
+ * This file is part of BEdita: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * See LICENSE.LGPL or <http://gnu.org/licenses/lgpl-3.0.html> for more details.
+ */
+namespace BEdita\Core\Shell;
+
+use BEdita\Core\Utility\Database;
+use Cake\Cache\Cache;
+use Cake\Console\Shell;
+use Cake\Core\Configure\Engine\JsonConfig;
+use Cake\Core\Plugin;
+use Cake\Filesystem\File;
+use Cake\ORM\TableRegistry;
+use Cake\Utility\Inflector;
+
+/**
+ * Basic shell commands:
+ *  - setup a new instance
+ *  - check instance
+ *  - internal cache cleanup
+ *
+ * @since 4.0.0
+ */
+class BeditaShell extends Shell
+{
+
+    /**
+     * {@inheritDoc}
+     */
```

Comment:
Since this method actually does nothing but adding a bit of configuration, I'd add a `@codeCoverageIgnore`, and maybe remove the `BeditaShellTest::testGetOptionParser()`.

We have already used this approach with Table classes (see `RolesTable::initialize()` for instance).